block-storage/src/main/scala/coop/rchain/blockstorage/dag/merging/data/DagMessage.scala

```diff
@@ -0,0 +1,15 @@
+package coop.rchain.blockstorage.dag.merging.data
+import coop.rchain.blockstorage.dag.Message
+
+final case class DagMessage[M](
+    id: M,
+    parents: Set[M],
+    fringe: Set[M],
+    // Cache of seen message ids
+    seen: Set[M]
+)
```

Comment:
This is the same object as Message. Is it really needed?
https://github.com/rchain/rchain/blob/7853b29923acb9d8b62e0f962ae3a6bf65173561/block-storage/src/main/scala/coop/rchain/blockstorage/dag/Finalizer.scala#L21-L33